### DISCUSSION TITLE
find location of maximum over a cube (preprocessing)

### DISCUSSION THREAD

**Discussion by rebeccaherman1** [ESMValGroup/ESMValCore#2163] [Ideas]
I see that it is simple to get the maximum of cube data over an area, or over a zonal or meridional region. But the resulting cube has only scalar spatial coordinates showing the middle of the region. How would I get a time series of the locations of the maximum values?

**Comment by valeriupredoi** [comment]
Hi @rebeccaherman1 you'd have to first get your desired maxima, then figure out what their locations in lat-lon space are by getting the indices (i, j) where these points sit in the 2D lat-lon data space, like this minimal example using Dask and lazy data:

```python
import numpy as np
import dask.array as da

lat = [11, 22]
lon = [33, 44]

# get the maximum
data = da.from_array([[33, 600], [22, 1000]])
x = da.where(data == da.max(data))

print(lat[x[0].compute()[0]], lon[x[1].compute()[0]])
```

- this returns a pair of (lat=22, lon=44)

`data` is the cube's `data` or `core_data()` when working with lazy Dask arrays
